map/src/components/OsmAndMap.js

```diff
@@ -69,9 +58,43 @@ async function addTrackToMap(ctx, file, map) {
     ctx.setGpxFiles(ctx.gpxFiles);
 }
 
-function removeTrackFromMap(file, map) {
-    map.current.removeLayer(file.gpx);
-    file.gpx = null;
+function removeLayerFromMap(file, map) {
+    if (file && file.gpx && map.current.hasLayer(file.gpx)) {
+        map.current.removeLayer(file.gpx);
+        file.gpx = null;
+    }
+}
+
+async function getFileData(file) {
+    let trackData;
+    if (file.url.substr(0, 1) === '<') { // direct XML has to start with a <
+        trackData = file.url;
+    } else {
+        let response = await Utils.fetchUtil(file.url, file.urlopts ? file.urlopts : {});
+        if (response.ok) {
+            trackData = await response.text();
+        } else {
+            trackData = '<gpx version="1.1" />'
+        }
+    }
+    return trackData;
+}
+
+async function addFavoritesToMap(ctx, file, map) {
```

Comment:
Likely it should be refactored and Tracks + Favorites should be in separate layers https://github.com/osmandapp/web/tree/e280e0d8997736a08d8d0c93b9efdd1db8c4a83a/map/src/components/layers